src/Form/LegacyConsumer/resources/js/conditinal-fields.js

```diff
@@ -38,21 +38,25 @@ document.addEventListener('readystatechange', event => {
 		donationForm.querySelectorAll('[data-field-visibility-conditions]').forEach(function (inputField) {
 			const visibilityConditions = JSON.parse(inputField.getAttribute('data-field-visibility-conditions'));
 			const visibilityCondition = visibilityConditions[0]; // Currently we support only one visibility condition.
-			const {field} = visibilityCondition;
 			let fieldSelector = getFieldSelector(inputField);
+			let {field} = visibilityCondition;
 
+			// Get field. It will tell use real name of field.
+			field = document.querySelector(`[name="${field}"], [name="${field}[]"]`);
```

Comment:
Is there any possibility that an input that matches the first selector (e.g. `<input name="food">`) _and_ an input that matches the second selector (e.g. `<input name="food[]">`) could exist in a form? Because if there is, this will only select the first one it finds. If there’s other logic that prevents this (like this function only runs for fields created by the FIelds API), then this isn’t an issue and you can ignore this.

Reply:
@knowler this logic is only for field API. As per you example, common field name values will be overwrite each other, so does not make sense to use them in the form.

Reply:
Awesome.